### PR TITLE
Event hdfs downloads

### PR DESCRIPTION
**Pull request by fmendezh**
HDFS/Hive processing was generalised to produce occurrence and events records suitable for GBIF downloads
This depends on branches:

- [https://github.com/gbif/dwc-api/tree/event-hdfs-downloads](https://github.com/gbif/gbif-api/tree/event-hdfs-downloads)
- [https://github.com/gbif/dwc-api/tree/event-hdfs-downloads](https://github.com/gbif/dwc-api/tree/event-hdfs-downloads)
- [https://github.com/gbif/postal-service/tree/event-hdfs-downloads](https://github.com/gbif/postal-service/tree/event-hdfs-downloads)

https://github.com/gbif/pipelines/issues/738

### CODE REVIEW

File: sdks/models/src/main/avro/occurrence-hdfs-record.avsc [hunk 1]
```diff
@@ -734,6 +734,26 @@
...
+  }, {
+    "name" : "datasettitle",
+    "type" : [ "string", "null" ]
+  }, {
+    "name" : "parenteventgbifid",
+    "type" : [ {
+      "type" : "array",
+      "items" : [ {
+        "type" : "record",
+        "name" : "ParentEventGbifId",
```
Comment (marcos-lg) [inline sdks/models/src/main/avro/occurrence-hdfs-record.avsc:746]: shouldn't this be in lower case too? or no need?

File: gbif/coordinator/tasks/src/main/java/org/gbif/pipelines/tasks/occurrences/hdfs/OccurrenceHdfsViewCallback.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      MessagePublisher publisher,
+      CuratorFramework curator,
+      PipelinesHistoryClient historyClient,
+      ExecutorService executor) {
+    return new OccurrenceHdfsViewCallback(config, publisher, curator, historyClient, executor);
+  }
+
+  @Override
+  public String routingKey() {
+    return PipelinesHdfsViewBuiltMessage.ROUTING_KEY;
```
Comment (marcos-lg) [inline gbif/coordinator/tasks/src/main/java/org/gbif/pipelines/tasks/occurrences/hdfs/OccurrenceHdfsViewCallback.java:44]: I think this should be as it was before:

```
String routingKey =
        new PipelinesInterpretedMessage().setRunner(config.processRunner).getRoutingKey();
```

These clis are deployed with 2 different configs (one for standalone and the other for distributed) and they use different routing keys. It'd have to be the same for the `EventsOccurrenceHdfsViewCallback`

File: sdks/models/src/main/avro/occurrence-hdfs-record.avsc (hunk 1 above)
Comment (fmendezh) [reply sdks/models/src/main/avro/occurrence-hdfs-record.avsc:746]: this schema is generated from the occurrence project, I'll look to see how easy is change that to provide better names for record types

File: sdks/core/src/main/java/org/gbif/pipelines/core/io/AvroReader.java [hunk 3]
```diff
@@ -162,6 +162,9 @@ private static <T extends Record> Map<String, T> readRecords(
   private static List<Path> parseWildcardPath(FileSystem fs, String path) {
     if (path.contains("*")) {
       Path pp = new Path(path).getParent();
+      if (!fs.exists(pp)) {
+        return Collections.emptyList();
+      }
```
[the remark this answers is not in the document]
Comment (fmendezh) [reply sdks/core/src/main/java/org/gbif/pipelines/core/io/AvroReader.java:167]: throwing an exception broke the execution processing archives of events and occurrences and one of them were empty

**Pull request merged**